google/cloud/bigquery/client.py

```diff
@@ -3524,7 +3515,7 @@ def query_and_wait(
         api_timeout: TimeoutType = DEFAULT_TIMEOUT,
         wait_timeout: Union[Optional[float], object] = POLLING_DEFAULT_VALUE,
         retry: retries.Retry = DEFAULT_RETRY,
-        job_retry: retries.Retry = DEFAULT_JOB_RETRY,
+        job_retry: Optional[retries.Retry] = DEFAULT_JOB_RETRY,
```

Comment:
The warning only applies to `query()` since we don't take a `job_id` in `query_and_wait`. We can revert this change.

```suggestion
        job_retry: retries.Retry = DEFAULT_JOB_RETRY,
```